best-practices-in-kubernetes-security.md

```diff
@@ -0,0 +1,180 @@
+
+**Load balancing:** Kubernetes support load balancing and network traffic distribution to ensure that the deployment is reliable.
+
+**Scalability:** Kubernetes enables your application to be scaled both vertically and horizontally. Since containers are lightweight by design, you can build them at the drop of a hat and then scale your application quickly.
+
+**Vendor and Cloud agnostic:** Kubernetes is vendor and Cloud neutral. It can run on Microsoft Azure, Amazon Web Services (AWS), Google Cloud Platform (GCP), or even on-premise.
+
+## **The Kubernetes Control Plane and its Components**
+
+The Kubernetes Control Plane a.k.a. the Master Node, serves as the brain of the cluster that manages the worker nodes, and it is responsible for ensuring that the system is up and running and is in perfect working order. It serves as the primary entry point for administrators and users to manage the different nodes in the cluster. The control plane comprises the Kubernetes API server, etcd, and several controllers.
+
+The following are the purposes of each of the components of the control plane:
+
+-   etcd - This is the fundamental component of a Kubernetes cluster used to store configuration data such that it is accessible to each node in a Kubernetes cluster.
+
+-   kube-apiserver - This is a REST-based interface through which all communications pass and controls all management and operational functions in Kubernetes.
+
+-   kube-controller-manager - This is a service that is responsible for keeping a close eye on the state of a cluster.
+
+-   kube-scheduler - This component is accountable for scheduling cluster workloads.
+
+The malicious attackers might try to gain access to the Kubernetes API Server and Control Plane. Once they have been hacked, the attackers may compromise the entire cluster by altering the pods. They can deploy new ones or modify or even delete the existing pods from the cluster.
+
+## **Kubernetes Security: Understanding the Challenges**
+
+While Kubernetes is a popular technology for delivering containerized apps, scaling Kubernetes environments is challenging. Each new container increases the attack surface. To be able to effectively deal with Kubernetes security, it is imperative that you’ve complete visibility into every managed container and application request.
+
+Kubernetes is great at orchestration but not as adept at security. For all deployments, it is thus imperative to employ the appropriate deployment architecture and security best practices. In the sections that follow we’ll take a deep dive into some of the Kubernetes security best practices.
+
+### **Securing the Control Plane**
+
+Integrity monitoring on the most critical Kubernetes files is a vital first step to securing the control plane. This way, you will get an instant notification if the settings of your Kubernetes environment are altered in any way. Note that critical files are those that, if hacked or compromised, might cause the whole cluster to fail.
```

Comment:
how would a user do that?  Could you include an example, or link to a guide ?

Reply:
This can be achieved by implementing detection policies to monitor any suspicious filesystem changes on critical Kubernetes files such as the following:

Creation or removal of files or directories
Renaming of files or directories
Any changes to file or directory security settings such as permissions, ownership, etc
Any changes made to the files pertaining a container

Here's a good article on this: https://www.trendmicro.com/vinfo/us/security/news/virtualization-and-cloud/the-basics-of-keeping-your-kubernetes-cluster-secure-part-1